plugins/wacom-usb/fu-wac-module-sub-cpu.c

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2024 Jason Gerecke <jason.gerecke@wacom.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-wac-common.h"
+#include "fu-wac-device.h"
+#include "fu-wac-module-sub-cpu.h"
+#include "fu-wac-struct.h"
+
+struct _FuWacModuleSubCpu {
+	FuWacModule parent_instance;
+};
+
+G_DEFINE_TYPE(FuWacModuleSubCpu, fu_wac_module_sub_cpu, FU_TYPE_WAC_MODULE)
+
+#define FU_WAC_MODULE_SUB_CPU_PAYLOAD_SZ      256
+#define FU_WAC_MODULE_SUB_CPU_START_NORMAL    0x00
+
+typedef struct __attribute__((__packed__)) {
+	guint32 addr;
+	guint32 cdata_len;
+	guint8 cdata[FU_WAC_MODULE_SUB_CPU_PAYLOAD_SZ];
+} FuWacModuleSubCpuBlockData;
```

Comment:
this needs to be defined in the .rs file please

Reply:
Realized I could just use FuChunk instead of this custom struct. New version will use that instead.